src/parser.coffee

```diff
@@ -257,6 +306,8 @@ exports.Parser = class Parser
     stack = []
     document = new model.Document(); head = document.start
 
+    currentlyCommented = false
```

Comment:
applyMarkup is a pretty long method now.  At the top of it, add a comment describing (1) what the job of the method is, (2) what the hard things are that need to be tracked and (3) a brief overview of the implementation strategy.